### ISSUE TITLE
Add support for run only once

### ISSUE THREAD

**Issue by dadoonet** [dadoonet/fscrawler#227]
Sometimes you would like to control externally (using a crontab) for example, how often you want to scan an index a hard drive.
Also, you might want to do it only once so you don't need to "watch" for changes.

In such a case, waiting for a message in logs is far from being ideal.

Let's add a new option `loop` which defines the number of runs we want before exiting:

* `X` where X is a negative value means infinite, like `-1` (default)
* `0` means that we don't run FSCrawler. Well... Not sure what could be the usage. May be checking that the configuration file is correct?
* `X` where X is a positive value is the number of runs before it stops.

If you want to scan your hard drive only once, run with `--loop 1`.

Based on https://github.com/dadoonet/fscrawler/issues/223#issuecomment-245300835

**Comment by Kangly** [comment]
hello,
Why not completely terminate the process?

**Comment by dadoonet** [comment]
WDYM?

**Comment by Kangly** [comment]
localhost:fscrawler-2.2-SNAPSHOT kangly$ bin/fscrawler test_type1 --loop 1
16:10:50,876 INFO  [f.p.e.c.f.FsCrawlerImpl] Starting FS crawler
16:10:51,203 INFO  [f.p.e.c.f.FsCrawlerImpl] FS crawler started for [test_type1] for [/Users/kangly/Downloads/report] every [5m]
16:10:52,223 INFO  [f.p.e.c.f.FsCrawlerImpl] FS crawler is stopping after 1 run

Why not direct close?
Forced closure process 
I had to use  ctrl+c

**Comment by Kangly** [comment]
I'm sorry.
My English communication level is general.

**Comment by dadoonet** [comment]
@Kangly No worries.

I see. So it looks like a bug, right?

Can you confirm that? If so can you open another issue like "FSCrawler does not stop when using loop parameter"?

**Comment by Kangly** [comment]
No, it stopped.
But,
The command: bin/fscrawler job_name --loop 1 
Why did not return to the command line input mode?
For example, 
localhost:fscrawler-2.2-SNAPSHOT kangly$ 

I can directly write other commands.

I'm sorry. I think this problem when used in PHP. I want to call this command.

May my idea is wrong.

**Comment by dadoonet** [comment]
So you wanted it to start and exit immediately but still run in the background? Is so, this is not possible.
I have no experience in PHP so I can't really tell.

BTW which other language do you speak?

**Comment by Kangly** [comment]
I wanted it to exit immediately.
Don't run in the background .
I will start it again when needed.

Chinese is my mother tongue. Other languages? I can only speak a little English.

Maybe I need to solve the problem from the PHP, I am doing.

Thank you for helping me.

**Comment by dadoonet** [comment]
> I wanted it to exit immediately.

FSCrawler will exit immediately after it indexed all documents available within your directory.
So I don't understand what is exactly the problem.

> Chinese is my mother tongue. 

Ok. You can ask here in Chinese and I'll ask @medcl to help me sorting this out.

**Comment by medcl** [comment]
happy to help here @dadoonet
 @Kangly 你可以说中文的哈

**Comment by jberkenbilt** [comment]
I am also observing that it doesn't exit.  I will create a separate ticket.

**Comment by jberkenbilt** [comment]
See #266
